utils/src/main/java/software/amazon/awssdk/utils/builder/ToCopyableBuilder.java

```diff
@@ -29,4 +31,18 @@
      * @return a builder for type T
      */
     B toBuilder();
+
+    /**
+     * A convenience method for calling {@link #toBuilder()}, updating the returned builder and then calling
+     * {@link CopyableBuilder#build()}. This is useful for making small modifications to the existing object.
+     *
+     * @param modifier A function that mutates this immutable object using the provided builder.
+     * @return A new copy of this object with the requested modifications.
+     */
+    default T modify(Consumer<B> modifier) {
```

Comment:
FYI both `Kotlin` and `Scala` have this convenience method on their data-classes and they call it `copy`

https://alvinalexander.com/scala/scala-case-class-copy-method-example
https://kotlinlang.org/docs/reference/data-classes.html